src/python/pants/backend/helm/target_types.py

```diff
@@ -0,0 +1,103 @@
+# Copyright 2022 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import annotations
+
+import os
+from abc import ABCMeta
+from dataclasses import dataclass
+from textwrap import dedent
+
+from pants.core.goals.package import OutputPathField
+from pants.engine.target import (
+    COMMON_TARGET_FIELDS,
+    BoolField,
+    Dependencies,
+    FieldSet,
+    MultipleSourcesField,
+    SingleSourceField,
+    Target,
+    TriBoolField,
+)
+from pants.util.docutil import bin_name
+
+# -----------------------------------------------------------------------------------------------
+# Generic commonly used fields
+# -----------------------------------------------------------------------------------------------
+
+
+class HelmSkipPushField(BoolField):
+    alias = "skip_push"
+    default = False
+    help = f"If set to true, do not push this helm chart to registries when running `{bin_name()} publish`."
+
+
+# -----------------------------------------------------------------------------------------------
+# `helm_chart` target
```

Comment:
Does the `Chart.lock` file need to be known to Pants at all?

Reply:
So far we have avoided dealing with the `Chart.lock` file but some sort of implementation would be needed.

This puts forward the question of whether we want Pants to manage Helm's 3rd party dependencies or not. In Helm, the `dependencies` of a given chart are always _3rd party_, meaning that they are supposed to come from published repositories while first party dependencies (subcharts) are meant to be put inside the `charts/` folder inside the given chart.

In our experience this is somewhat messy, specially if you have a few customs charts of `library` type which are meant to be used by other, more complex, charts as it forces you to go through the `package->publish->generate-lockfiles` flow whenever you make a change to the smaller charts.

In our implementation what we do is parsing `Chart.yaml` file and infer any dependencies from it, whether they are first- or third- party, then in the rule logic that returns `HelmChart` type, we package all those subcharts under the `charts/` folder and re-render the `Chart.yaml` including the first party dependencies in it.

With that approach we avoid of publishing the first party dependencies before being able to run any `lint` or `test` goals in the other charts that reuse those.

In theory it should be possible to find a solution that satisfies both, like copying into the `charts/` folder any first party dependencies and then use `helm dependency update`/`helm dependency build` commands to let Helm resolve the third party dependencies based on the Chart.yaml.